### DISCUSSION TITLE
tweaks to date field

### DISCUSSION THREAD

**Discussion by terrafrost** [Laravel-Backpack/community-forum#1100] [Ideas]
So right now on the [date field](https://backpackforlaravel.com/docs/6.x/crud-fields#date-1) the format is fixed as mm/dd/yyyy. It'd be nice if this could be easily changed through an option.

Also, let's say you've typed in the month and are now in the day section of the date and then suddenly realized that you misstyped in the month - maybe you typed in 06 instead of 07 or whatever. Your first instinct would probably be to hit backspace but, oh wait, that doesn't work - it'd be nice if it did.

Additionally maybe the length of the yyyy field could be limited to 4 characters? I mean if someone really wants to be able to enter 6 digit years then maybe that behavior could be enabled through the use of a new option?

Finally, for most fields you can just hit tab to skip over them and go to the next field. But not for date fields. You have to hit tab not once but four times to get to the next field. Like you hit tab to get into a date field but that puts you into the mm part of the date. You hit tab again and then you get put into the dd part of the date. You hit tab once more and you get put into the yyyy part of the date. And then you hit tab one more time and you go to the calendar icon. Hitting tab after that will take you to the next field but you have to hit tab a lot to do so - more than for any other field that I'm at least using. It'd be nice if hitting tab once on that field would skip the whole thing. I mean, does that field even accept partial dates? If not then why would you want to tab into the middle part of the date? And the calendar dialog really doesn't make sense with tabbing, anyway - if you want to enter in the date that way you'd be better off just using the mouse.

**Comment by pxpm** [comment]
Hello @terrafrost 

The `date` field is just a regular html date field, you can check the field options and/or configurations here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date

Cheers

**Comment by karandatwani92** [comment]
You can also try [date_picker](https://backpackforlaravel.com/docs/6.x/crud-fields#date_picker-pro) for more possible customization.
